app/src/main/kotlin/com/intive/picover/common/viewmodel/StateFulViewModel.kt

```diff
@@ -0,0 +1,12 @@
+package com.intive.picover.common.viewmodel
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import com.intive.picover.common.viewmodel.state.ViewModelState
+
+abstract class StateFulViewModel<T> constructor(initialState: ViewModelState<T> = ViewModelState.Loading) : ViewModel() {
+
+	protected val _state = mutableStateOf(initialState)
+	val state: State<ViewModelState<T>> = _state
+}
```

Comment:
Probably all or almost all view models will need a state. Instead of repeating the declaration in every view model it's reasonable to use basic class here. 
`state` declaration is especially loud and broad in case of List, check below example
```
	val state: MutableState<ViewModelState<List<String>>> = mutableStateOf(ViewModelState.Loading)
```
of course explicit type can be dropped, but we would need to extend `mutableStateOf` with `ViewModelState<List<String>>` type in the following way due to type inference:
```
	val state = mutableStateOf<ViewModelState<List<String>>>(ViewModelState.Loading)
```

Using `StateFulViewModel` simplifies this via replacing inheritance from `: ViewModel` to `StateFulViewModel<List<String>>` 
Additionally, introduced backing field (`_state`) would prevent from setting up a state, outside of the view model, without any duplicated declarations. That also opens a possibility to create some extensions to handle some responses from repositories (`kotlin.Result`, `Flow`) in a generic way.